### PR TITLE
Reales notes fix bugs

### PR DESCRIPTION
**Pull request by omerKarkKatz**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5848
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6176

## Description
Added fixes to open dev-bugs. 
1. validate that get yml return value is of type dict. 
2. validate that no new release notes are being generated when running update-release-notes on a first version

### CODE REVIEW

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -920,6 +920,9 @@ def get_from_version(file_path):
         get_yaml(file_path) if file_path.endswith("yml") else get_json(file_path)
     )
 
+    if not isinstance(data_dictionary, dict):
+        raise ValueError("yml file returned is not of type dict")
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:924]: do we not have `yaml`s that are lists?

Comment (omerKarkKatz) [reply demisto_sdk/commands/common/tools.py:924]: I dont know of such a case and the next line in this function is data_dictionary.get("fromversion") leading to an error.

Comment (dorschw) [reply demisto_sdk/commands/common/tools.py:924]: true

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,8 @@
 * Added the `DEMISTO_SDK_GRAPH_FORCE_CREATE` environment variable. Use it to force the SDK to recreate the graph, rather than update it.
 * Added support for code importing multi-level ApiModules to **lint**.
 * Added a validation that the **modeling-rules test** command will fail if no test data file exist.
+* Added a better informative message when yml file is not parsed as a dict **common/tools.py**.
```
Comment (dorschw) [inline CHANGELOG.md:19]:
```suggestion
* Improved the clarity of error messages for cases where yml files cannot be parsed as a dictionary.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,8 @@
...
+* Fixed an issue where **update-release-notes** would generate a new file for first version.
```
Comment (dorschw) [inline CHANGELOG.md:20]:
```suggestion
* Fixed an issue where **update-release-notes** generated release notes for packs in their initial version (1.0.0).
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 4]
```diff
@@ -543,7 +549,7 @@ def get_pack_metadata(self) -> dict:
             data_dictionary = get_json(self.metadata_path, cache_clear=True)
         except FileNotFoundError as e:
             raise FileNotFoundError(
-                f"Pack {self.pack} was not found. Please verify the pack name is correct."
+                f"Pack {self.pack} metadata file, was not found. Please verify the pack name is correct, and that the file exsits."
```
Comment (dorschw) [inline demisto_sdk/commands/update_release_notes/update_rn.py:552]:
```suggestion
                f"The metadata file of pack {self.pack} was not found. Please verify the pack name is correct, and that the file exsits."
```

**Review by dorschw — APPROVED** [review]

**Pull request merged**